src/tfvc/commands/delete.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+"use strict";
+
+import { TeamServerContext} from "../../contexts/servercontext";
+import { IArgumentProvider, IExecutionResult, ITfvcCommand } from "../interfaces";
+import { ArgumentBuilder } from "./argumentbuilder";
+import { CommandHelper } from "./commandhelper";
+
+/**
+ * This command deletes the files passed in.
+ * It returns a list of all files marked for deletion.
+ * delete /detect [/lock:none|checkin|checkout] [/recursive]
+ * delete [/lock:none|checkin|checkout] [/recursive] <itemSpec>...
+ */
+export class Delete implements ITfvcCommand<string[]> {
+    private _serverContext: TeamServerContext;
+    private _itemPaths: string[];
+
+    public constructor(serverContext: TeamServerContext, itemPaths: string[]) {
+        CommandHelper.RequireStringArrayArgument(itemPaths, "itemPaths");
+        this._serverContext = serverContext;
+        this._itemPaths = itemPaths;
+    }
+
+    public GetArguments(): IArgumentProvider {
+        return new ArgumentBuilder("delete", this._serverContext)
+            .AddAll(this._itemPaths);
+    }
+
+    public GetOptions(): any {
+        return {};
+    }
+
+    //Delete returns either 0 (success) or 100 (failure).  IF we fail, simply throw.
+    public async ParseOutput(executionResult: IExecutionResult): Promise<string[]> {
+        let lines: string[] = CommandHelper.SplitIntoLines(executionResult.stdout, false, true /*filterEmptyLines*/);
+
+        if (executionResult.exitCode === 100) {
+            CommandHelper.ProcessErrors(this.GetArguments().GetCommand(), executionResult, true);
```

Comment:
Seems like this should come first. No need to split the lines if this throws. The exit code check also seems unnecessary if you only expect 100 or 0.